arangod/Aql/ShardLocking.h

```diff
@@ -0,0 +1,117 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2016 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Michael Hackstein
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef ARANGOD_AQL_SHARD_LOCKING_H
+#define ARANGOD_AQL_SHARD_LOCKING_H 1
+
+#include "Cluster/ClusterInfo.h"
+#include "VocBase/AccessMode.h"
+
+#include <set>
+
+namespace arangodb {
+namespace velocypack {
+class Builder;
+}
+
+namespace aql {
+
+struct Collection;
+class ExecutionNode;
+class Query;
+
+class ShardLocking {
```

Comment:
I know I personally would appreciate some class comments for the new classes, giving a rough idea how the class fits into the overall architecture, how it's used, etc.

Reply:
Agree!